### PR TITLE
Implemented initial draft of Scatter Plot

### PR DESCRIPTION
**Pull request by dalerasrorov-eb**
Implemented Scatter Plot, initial work.

## Description
* Implemented a new chart - Scatter Plot. The API functions available to the user are quite similar to the existing charts. The unique one is  `maxCircleArea` which allows the user to increase the maximum area scale range (basically increasing circle radius).
* So far it has no interactivity or tooltip, but it already has a `customClick` with action dispatcher that can be extended to other events. Will add tooltip 
* Our designer Sun added a few comments that I am going to address in the next commits. This commit is quite large already, so we will add more features bit by bit and frequently.
* Added one new demo
* Added data API and documentation to the API docs and followed styleguide

## Motivation and Context

https://github.com/eventbrite/britecharts/issues/126

## How Has This Been Tested?
Added a new test file `scatter-plot.spec.js` that tests the basic functionalities and render expectations as well as `customClick` handler and margin object test.

## Screenshots (if appropriate):
screen shot 2018-03-20 at 1 39 04 am

## Types of changes

- [ ] Refactor (changes the way we code something without changing its functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the [code style guide](https://github.com/eventbrite/britecharts/blob/master/CODESTYLEGUIDE.md) of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: demos/demo-scatter-plot.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+const scatterPlot = require('./../src/charts/scatter-plot');
+const colors = require('./../src/charts/helpers/color');
+const dataBuilder = require('./../test/fixtures/scatterPlotDataBuilder');
+
+const aTestDataSet = () => new dataBuilder.ScatterPlotDataBuilder();
+
+require('./helpers/resizeHelper');
+
+function createSimpleScatterPlot() {
+        let scatter = scatterPlot();
```
Comment (Golodhros) [inline demos/demo-scatter-plot.js]: Nit: this indentation seems off a bit

File: demos/demo-scatter-plot.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        if (containerWidth) {
+            dataset = aTestDataSet().withFourNames().build();
+
+            scatter
+                .width(750)
+                .maxCircleArea(15);
+
+            scatterPlotContainer.datum(dataset).call(scatter);
+        }
+}
```
Comment (Golodhros) [inline demos/demo-scatter-plot.js]: Should we ask Sun for a loading state for this chart? I don't think we have something even remotely close to it...

File: test/json/scatterDataWithFourNames.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{
+    "data": [
+        {
+            "name": "Nabisco",
```
Comment (Golodhros) [inline test/json/scatterDataWithFourNames.json]: I have a list of names to add in the demos, all of them pretty brite:

Comment (Golodhros) [reply test/json/scatterDataWithFourNames.json]: radiating
reflecting
shining
sunshine
vivid
brilliant
blazing
brilliant
dazzling
flashing
glistening
glittering
intense
luminous
radiant
shimmering
shiny
silvery
sparkling
sunny
vivid

File: test/specs/scatter-plot.spec.js [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+        afterEach(() => {
+            containerFixture.remove();
+            f = jasmine.getFixtures();
+            f.cleanUp();
+            f.clearCache();
+        });
+
+        it('should render a chart with minimal requirements', () => {
+            expect(containerFixture.select('.scatter-plot').empty()).toBeFalsy();
```
Comment (Golodhros) [inline test/specs/scatter-plot.spec.js]: I really want us to move away from this format on the tests and start doing something like:
  let expected = 1;
  let actual = containerFixture.select('.scatter-plot').nodes().length;

  expect(actual).toEqual(expected);

No rush for it, as the rest are like this one...

File: src/charts/scatter-plot.js [hunk 5]
```diff
@@ -0,0 +1,577 @@
+import { colorSchema, nameLabel } from './bar';
+import { xAxisLabel } from './line';
```
Comment (Golodhros) [inline src/charts/scatter-plot.js]: Surprised by this two up here!

File: src/charts/scatter-plot.js [hunk 6]
```diff
@@ -0,0 +1,577 @@
...
+                chartHeight = height - margin.top - margin.bottom;
+
+                buildScales();
+                buildSVG(this);
+                buildAxis();
+                drawAxis();
+                drawGridLines();
+                drawDataPoints();
+
+                // TODO: the rest of the functions
```
Comment (Golodhros) [inline src/charts/scatter-plot.js]: Is this needed?

File: src/charts/scatter-plot.js [hunk 7]
```diff
@@ -0,0 +1,577 @@
...
+
+            // TODO: build the rest of the container groups
+            // Build clip container for the inner chart part
+        }
+
+        /**
+         * Draws the x and y axis on the svg object within their
+         * respective groups along with their axis labels
+         * @private
+        */
```
Comment (Golodhros) [inline src/charts/scatter-plot.js]: Nit: alignment

File: src/charts/scatter-plot.js [hunk 8]
```diff
@@ -0,0 +1,577 @@
...
+         * @return void
+         */
+        function drawGridLines() {
+            svg.select('.grid-lines-group')
+                .selectAll('line')
+                .remove();
+
+            drawVerticalGridLines();
+            drawHorizontalGridLines();
+            drawHorizontalExtendedLine();
```
Comment (Golodhros) [inline src/charts/scatter-plot.js]: Aren't the grid lines optional in this one?

File: demos/demo-scatter-plot.js (hunk 2 above)
Comment (dalerasrorov-eb) [reply demos/demo-scatter-plot.js]: @Golodhros that would be a great idea! Once it's ready we can easily add it to the chart.

**Pull request merged**